### DISCUSSION TITLE
Add ultrasonic sensor for pellets tank level

### DISCUSSION THREAD

**Discussion by alextoro82** [philibertc/micronova_controller#52] [Ideas]
Hi Philibertc
First of all, many thanks for your contribution and guideline because you help me a lot.
I,m waiting some materials to connect my Edilkamin Pilot pellet stove that uses micronova board with my HA.
As add-on, I think we can add in the same D1 mini board an Ultrasonic Distance Sensor (HC-SR04) to know the level of pellets in the tank. 
With this sensor you can know the distance from the top of the tank and create some automations.
I will try to add it over your code

**Comment by basrihsn** [comment]
Why are you mentioning me with your scam links? Report this user

**Comment by olivimages** [comment]
> Hi Philibertc First of all, many thanks for your contribution and guideline because you help me a lot. I,m waiting some materials to connect my Edilkamin Pilot pellet stove that uses micronova board with my HA. As add-on, I think we can add in the same D1 mini board an Ultrasonic Distance Sensor (HC-SR04) to know the level of pellets in the tank. With this sensor you can know the distance from the top of the tank and create some automations. I will try to add it over your code

Hi,
Did you implement the sensor ? I'm also interested in upgrading with a distance sensor and the code you use to complete the one of  Philibertc

**Reply by alextoro82** [reply]
Yes, it´s not so complex.
In HA with ESPHome this is the code for the ultrasonic distance sensor HC-SR04. 
I converted the distance data received from ultrasonic sensor to the corresponding % of the pellet stove hopper
Hint replace 0.46 by the height of your pellet stove hopper, from the sensor to the bottom.

  - platform: ultrasonic
    trigger_pin: GPIO13 #D7 pin of ESP d1mini
    echo_pin: GPIO15 #D8 pin of ESP d1mini
    name: "Ultrasonic Sensor"
    update_interval: 60s # time for every update
    icon: "mdi:sack-percent"
    accuracy_decimals: 0      
    filters:
    - lambda: return (0.46-x)*(100/0.46);
    unit_of_measurement: "%"

**Reply by alextoro82** [reply]
If you are interested in HA ESPHome the micronova board used in the pellet stove is also supported (this is somehow linked with the Philibertc works)
Some info here https://esphome.io/components/micronova.html
